src/views/Home/Home.js

```diff
@@ -104,50 +104,85 @@ import Form from './components/Form'
 
 import JourneySideBar from 'components/JourneySideBar';
 
+import axios from 'axios'
+import { Category, ConstructionRounded } from '@mui/icons-material';
+
 const Home = () => {
-  const theme = useTheme();
-  const [shouldRenderForm, setShouldRenderForm] = useState(false)
-  
-  const addExperience = () => {
-    setShouldRenderForm(true)
-    // call api
+  const [experiences, setExperiences] = useState([])
+
+  useEffect(async () => {
+    const userData = JSON.parse(window.localStorage.getItem('user'))
+    const res = await axios.get(`/api/user/${userData._id}/experience`)
+    
+    setExperiences(res.data)
+  }, [])
+
+  console.log(experiences)
+
+  class Category {
+    constructor(name) {
+      this.name = name
+      this.experiences = []
+    }
   }
 
-  const categories = [
-    {
-      name: 'HTML',
-      experiences: [
-        {
-          name: 'experience 1',
-          preview: 'this is a descriptsadfsadfdsafasdfasfasdion/preview of the details of the experience'
-        },
-        {
-          name: 'experience 2',
-          preview: 'this is a descriptsadfsadfdsafasdfasfasdion/preview of the details of the experience'
+  function getAllCategories() {
+    const categories = []
+    experiences.forEach((experience, index) => {
+      const category = new Category(experience.category)
+      if (!(category in categories)) {
+        categories.push(category)
+        console.log(category.name)
+
+      }
+    })
+
+    categories.forEach((category, i) => {
+      experiences.forEach((experience, j) => {
+        if (experience.category == category.name) {
+          category.experiences.push(experience)
         }
-      ]
-    },
-    {
-      name: 'C++',
-      experiences: [
-        {
-          name: 'experience 1',
-          preview: 'this is a descriptsadfsadfdsafasdfasfasdion/preview of the details of the experience'
-        },
-        {
-          name: 'experience 2',
-          preview: 'this is a descriptsadfsadfdsafasdfasfasdion/preview of the details of the experience'
-        },
-        {
-          name: 'experience 3',
-          preview: 'this is a descriptsadfsadfdsafasdfasfasdion/preview of the details of the experience'
-        }
-      ]
-    }
-  ]
+      })
+    })
+    return categories
+  }
 
-  const drawerWidth = 240;
+  const categories = getAllCategories()
```

Comment:
What's the intended implementation for `getAllCategories()`?

Reply:
I ran into some problems with the categorization of the experiences. I dropped the function and forgot to remove it. Disregard

Reply:
Thanks for clarifying that.